### ISSUE TITLE
Best way to implement "Live" indicator

### ISSUE THREAD

**Issue by ppodgorski** [google/ExoPlayer#7711]
I need to implement such a feature. When user's playback is at the live edge, display "LIVE" label. Otherwise, display current position (or another "not live" indicator like on YouTube lives). What is the best way to do this? 

Thank you for any help :)

**Comment by swifityG** [comment]
I did it with by using `player.isCurrentWindowLive`  in `onPlayerStateChanged` and its work well

**Comment by ojw28** [comment]
The suggestion from @swifityG works well if your aim is to always show the indicator when playing live content, but not for on-demand content.

If your goal is that you're playing live content, and you want to show the indicator only if the user's playback is at the live edge, then I think you can probably just use `getCurrentPosition` (gets the current playback position in the live window) and `getDuration` (gets the duration of the live window), and show the live indicator if the current position is within some tolerance of the duration (i.e., the current position is near the live edge). Since live windows often update in a discrete steps, you will probably want to build a bit of hysteresis into your logic that determines whether the live indicator is being shown. This will allow you to avoid repeatedly toggling the indicator on and off if the user is playing back at around your specified tolerance behind the live edge.

**Comment by ppodgorski** [comment]
@ojw28 thanks for your reply. I've started my implementation only with `getDuration` and `getCurrentPosition` methods but it didn't work well. It turned out that I have to take into account the window.defaultPositionMs value. Now it works quite good.

There is one more thing. I've started with checking the live indicator state in `onSeekProcessed` callback, but live indicator should also change when user pause the playback. So I've ended up with repeatedly checking the live state with 1000ms interval (using Handler.postDelayed). Maybe there is a better way to do this?
